src/nlr/Layer.cpp

```diff
@@ -1481,4 +1501,142 @@ double Layer::getEliminatedNeuronValue( unsigned neuron ) const
     return _eliminatedNeurons[neuron];
 }
 
+void Layer::reduceIndexFromAllMaps( unsigned startIndex )
+{
+    // Adjsut the source layers
+    Map<unsigned, unsigned> copyOfSources = _sourceLayers;
+    _sourceLayers.clear();
+    for ( const auto &pair : copyOfSources )
+        _sourceLayers[pair.first >= startIndex ? pair.first - 1 : pair.first] = pair.second;
+
+    // Adjust all weight maps
+    adjustWeightMapIndexing( _layerToWeights, startIndex );
+    adjustWeightMapIndexing( _layerToPositiveWeights, startIndex );
+    adjustWeightMapIndexing( _layerToNegativeWeights, startIndex );
+
+    // Adjust the neuron activations
+    for ( auto &neuronToSources : _neuronToActivationSources )
+    {
+        for ( auto &source : neuronToSources.second )
+        {
+            if ( source._layer >= startIndex )
+                --source._layer;
+        }
+    }
+}
+
+void Layer::adjustWeightMapIndexing( Map<unsigned, double *> &map, unsigned startIndex )
+{
+    Map<unsigned, double *> copyOfWeights = map;
+    map.clear();
+    for ( const auto &pair : copyOfWeights )
+        map[pair.first >= startIndex ? pair.first - 1 : pair.first] = pair.second;
+}
+
+void Layer::reduceIndexAfterMerge (unsigned startIndex)
+{
+    if ( _layerIndex >= startIndex )
+    {
+        _layerIndex--;
```

Comment:
Always prefer --i to i--. I can explain why in person.
Also, no need to use { and } for such simple cases.